### PR TITLE
[ #158518876] Create Ride Offers

### PR DESCRIPTION
**Pull request by SnyderMbishai**
#### Introduction
This is a PR for creating the endpoint for adding rides.

#### Task description
Details of the ride can be saves after being created through the endpoint.
Before the this particular endpoint, the registration and login endpoints were created.
Afterwards, the tests for the endpoints were refactored to run.

#### Manual testing
- git clone the repo to a virtual environment
- pip install -r requirements.txt
- cd to the right branch
- flask run and then test the endpoints with postman

#### screenshot
screenshot from 2018-06-24 18-51-15

### CODE REVIEW

**Comment by meshack-mbuvi** [comment]
I believe `route` should be the roads driver is going to use. For instance, Mombasa Rd,Waiyaki way,and such

File: tests/test_api.py [hunk 1]
```diff
@@ -11,10 +11,32 @@ class Base(unittest.TestCase):
...
+    
+    def test_username_field_not_empty(self):
+
+        pass
+
+    def test_password_field_not_empty(self):
+        pass
+
 
     def tearDown(self):
```
Comment (Quantum-35) [inline tests/test_api.py]: in your tearDown you can clear your datastructure if its a list e.g 
j = []
j.clear()
to delete data after the test is done

**Comment by JoshuaOndieki** [comment]
I think your commit *03b7e11* contradicts with exactly what you did. Looking at it for the first time makes one wonder, *Why would you add cache files?*
Of course, you ignored the cache files from being tracked by adding the **file names** in your **.gitignore**
Note you added the file names and not the actual cache files.
Tip: Make the commit message to directly describe changes made.
TIPS :arrow_right:  https://github.com/erlang/otp/wiki/writing-good-commit-messages

**Comment by JoshuaOndieki** [comment]
According to my knowledge,
- add classes
- delete file
- update files
- add unimportant files
- update file
Are NOT valid commit messages

File: tests/test_api.py [hunk 2]
```diff
@@ -74,30 +78,30 @@ def test_successful_ride_creation(self):
...
+#         """  Method for testing a ride request is successfully placed.  """
         
-        response = self.client.post('api/v1/rides/1/requests', data=json.dumps(self.request_data), content_type='application/json')
-        result = json.loads(response.data)
-        self.assertEqual(response.status_code,201) 
-        self.assertEqual(result['message'],"The request has been made. Please wait for feedback from the driver.")
+#         response = self.client.post('api/v1/rides/1/requests', data=json.dumps(self.request_data), content_type='application/json')
+#         result = json.loads(response.data)
+#         self.assertEqual(response.status_code,201) 
+#         self.assertEqual(result['message'],"The request has been made. Please wait for feedback from the driver.")
```
Comment (JoshuaOndieki) [inline tests/test_api.py]: Remove unused lines of code.

File: tests/test_api.py [hunk 3]
```diff
@@ -1,22 +1,107 @@
...
+#     """  Class for testing ride requests.  """
+
+#     def test_request_to_join_a_ride(self):
+#         """  Method for testing a ride request is successfully placed.  """
         
+#         response = self.client.post('api/v1/rides/1/requests', data=json.dumps(self.request_data), content_type='application/json')
+#         result = json.loads(response.data)
+#         self.assertEqual(response.status_code,201) 
+#         self.assertEqual(result['message'],"The request has been made. Please wait for feedback from the driver.")
+
```
Comment (JoshuaOndieki) [inline tests/test_api.py]: A lot of unused code from line 83 to 106

File: api/views.py [hunk 4]
```diff
@@ -1,4 +1,121 @@
...
+        elif destination == " ":
+            return {'message':"provide the destination point"}
+        elif departure_time == " ":
+            return {'message':"please provide the departure time"}
+        elif route == " ":
+            return{'message':"What route will we be taking?"}
+        elif extras == " ":
+            return {'message':"Are you sure you have nothing else to add?"}
+        elif ride.route in [item.route for item in rides]:
+            return {'message':"Ride already exists, please create another"}
```
Comment (JoshuaOndieki) [inline api/views.py]: Consider a situation where almost all data points are not provided. Wouldn't it be wise returning a message about all the data not provided?

**Comment by TheSteelGuy** [comment]
Good @SnyderMbishai  also note that its so pythonic to always append a line at the end of code

**Pull request merged**